src/testdrive/src/lib.rs

```diff
@@ -114,7 +114,13 @@ async fn run_line_reader(
 
         for a in &actions {
             let redo = a.action.redo(&mut state);
-            redo.await.map_err(|e| PosError::new(e, a.pos))?
+            redo.await.map_err(|e| PosError::new(e, a.pos))?;
+            if state.skip_rest {
+                println!("skip-if query returned true; skipping rest of file");
+                break;
+            } else {
+                println!("skip-if query returned false; continuing");
+            }
```

Comment:
I think you want these `println!`s in the skip-if action itself to avoid printing on every remaining action, leaving just the `break` here.